hack/testing/check-log-speed.sh

```diff
@@ -0,0 +1,329 @@
+#!/bin/bash
+
+set -euo pipefail
+
+function get_running_pod() {
+    # $1 is component for selector
+    oc get pods -l component=$1 | awk -v sel=$1 '$1 ~ sel && $3 == "Running" {print $1}'
+}
+
+function query_es_from_es() {
+    oc exec $1 -- curl --connect-timeout 1 -s -k \
+       --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key \
+       https://localhost:9200/${2}*/${3}\?"$4"
+}
+
+# read JSON query arguments from stdin
+function query_es_from_es_json() {
+    oc exec -i $1 -- curl --connect-timeout 1 -s -k \
+       --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key \
+       https://localhost:9200/${2}*/${3} -d@-
+}
+
+function get_url_path_of_record() {
+    # returns $index/$type/$_id
+    python -c "import json, sys; hsh = json.loads(sys.stdin.read())['hits']['hits'][0]; print '%(_index)s/%(_type)s/%(_id)s'.encode('utf-8') % hsh"
+}
+
+function get_last_field_from_es() {
+    query_es_from_es $1 $2 _search "size=1\&sort=@{timefield}:desc" | \
+        get_field_value_from_json $3
+}
+
+function get_last_url_from_es() {
+    query_es_from_es $1 $2 _search "size=1&sort=${timefield}:desc${3:+&$3}" | \
+        get_url_path_of_record
+}
+
+function get_field_value_from_record() {
+    python -c 'import json, sys; print json.loads(sys.stdin.read())["_source"]["'"$1"'"].encode("utf-8")'
+}
+
+function get_field_value_from_es_url() {
+    oc exec -i $1 -- curl --connect-timeout 1 -s -k \
+       --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key \
+       https://localhost:9200/$2 | get_field_value_from_record "$3"
+}
+
+# $1 - shell command or function to call to test if wait is over -
+#      this command/function should return true if the condition
+#      has been met, or false if still waiting for condition to be met
+# $2 - shell command or function to call if we timed out for error handling
+# $3 - timeout in seconds - should be a multiple of $4 (interval)
+# $4 - loop interval in seconds
+function wait_until_cmd_or_err() {
+    let ii=$3
+    local interval=${4:-1}
+    while [ $ii -gt 0 ] ; do
+        $1 && break
+        sleep $interval
+        let ii=ii-$interval
+    done
+    if [ $ii -le 0 ] ; then
+        $2
+        return 1
+    fi
+    return 0
+}
+
+debug() {
+    if [ -n "${DEBUG:-}" ] ; then
+        echo "$@"
+    fi
+}
+
+verbose() {
+    if [ -n "${VERBOSE:-}" ] ; then
+        echo "$@"
+    fi
+}
+
+err() {
+    echo ERROR: "$@"
+}
+
+info() {
+    echo "$@"
+}
+
+info_same() {
+    verbose "$@"
+}
+
+info_diff() {
+    # $X is $N $units behind $Y
+    # or
+    # $X is up-to-date with $Y
+    if [ $2 = "0" -o $2 = "0.0" ] ; then
+        info_same $1 is up-to-date with $4 in $3
+    else
+        info $1 is $2 $3 behind $4
+    fi
+}
+
+es_pod=`get_running_pod es`
```

Comment:
it should work given all our ES pods are currently master-data nodes. this may change in the future if we can configure them differently but we can cross that bridge when we get to it